pages/api/user.js

```diff
@@ -72,52 +116,57 @@ export async function readUser(email) {
   });
 }
 
+/**
+ * Retrieves all users from Auth0.
+ * @returns {Promise}
+ */
 export async function allUsers() {
-  return prisma.user.findMany({
-    include: {
-      hospitalRole: true,
-      admin: true,
-    },
+  // Retrieve all the users in the username/password connection
+  const results = await managementClient.users.getAll({
+    q: "identities.connection:Username-Password-Authentication"
   });
+
+  return results.data.map((user) => { return {
+    email: user.email,
+    name: user.name,
+    admin: user.app_metadata.admin || false,
+    hospitalRole: user.app_metadata.hospitalRole || [],
+  }});
 }
 
 export async function allHospitalRoles() {
   return prisma.hospitalRole.findMany();
 }
 
-async function addData(req, res) {
-  const body = req.body;
-  const create = {
-    data: {
-      email: body.email,
-    },
-    include: {
-      hospitalRole: true,
-    },
-  };
-  console.log(
-    "Request body " +
-      JSON.stringify(body) +
-      " create value " +
-      JSON.stringify(create)
-  );
 
-  try {
-    const newEntry = await prisma.user.create(create);
-    return res.status(200).json(newEntry, { success: true });
-  } catch (error) {
-    console.log("Request error " + error);
-    res
-      .status(500)
-      .json({ error: "Error adding user" + error, success: false });
-  }
+/**
+ * Deletes a user from Auth0 by their user ID.
+ * @returns 204 No Content
+ */
+async function deleteUser(req, res) {
+  await managementClient.users.delete(req.body.id);
+  return res.status(204);
 }
 
-async function deleteData(req, res) {
-  const deleteConfirmation = await prisma.user.delete({
-    where: {
-      id: req.body.id,
-    },
-  });
-  return res.status(200).json(deleteConfirmation, { success: true });
+/**
+ * Extracts the users details from the session object
+ * @param ctx 
+ * @returns Null if the session is null, else user details
+ */
+export async function getUserFromSession(ctx) {
+  const session = await getSession(ctx.req, ctx.res);
+
+  // Session not active, redirect user to homepage
+  if (session === null) {
+    return null;
+  }
+
+  const userFromSession = session.user;
+  const userMetadata = userFromSession['https://vapartners.org/app_metadata'];
+  return {
+    email: userFromSession.email,
+    name: userFromSession.name,
+    admin: (userMetadata.va_partners.admin !== undefined) ? userMetadata.va_partners.admin : false,
```

Comment:
also check for userMetadata.va_partners != undefined

Reply:
Instead I'm going to throw an error before it hits this point if either the metadata or va_partners object is undefined.